experimental/oak_functions_with_envoy/client/client.yaml

```diff
@@ -0,0 +1,47 @@
+static_resources:
+  listeners:
+    # Accepts TCP streams and tunnels them over HTTP streams.
+    # https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/upgrades#tunneling-tcp-over-http
+    - name: client_listener
+      address:
+        # Client proxy address to listen on.
+        socket_address:
+          protocol: TCP
```

Comment:
Can it not listen over a domain socket or something else?

Reply:
It can. There is a [pipe](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/address.proto#envoy-v3-api-msg-config-core-v3-pipe) address type. But I'm not sure whether it will work nicely with TLS, since IIUC data that is sent to the socket is supposed to be on L7.

Reply:
makes sense, we should experiment with it at some point though